.cirrus.yml

```diff
@@ -60,112 +54,23 @@ meta_task:
     script: /usr/local/bin/entrypoint.sh
 
 
-validate_task:
-
-    container:
-        dockerfile: contrib/cirrus/Dockerfile
-
-    timeout_in: 20m
-
-    # TODO: Making this task as not-requred to help speed up development given a new CI system
-    allow_failures: $CI == $CI
-
-    env:
-        CONTAINER: true
-
-    # Avoid downloading this stuff every time
-    gocache_cache:
-        folder: "${GOCACHE}"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
-
-    # Avoid needless rebuilding of tooling binaries
-    gopath_cache:
-        folder: "${GOPATH}/bin"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
-
-    script:
-        - $SCRIPT_BASE/setup.sh
-        - $SCRIPT_BASE/validate.sh
-
-
+# make sure that the ./vendor directory is up to date
 vendor_task:
 
     container:
-        dockerfile: contrib/cirrus/Dockerfile
-
-    timeout_in: 20m
-
-    # TODO: Making this task as not-requred to help speed up development given a new CI system
-    allow_failures: $CI == $CI
-
-    env:
-        CONTAINER: true
-
-    # Avoid downloading this stuff every time
-    gocache_cache:
-        folder: "${GOCACHE}"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
-
-    # Avoid needless rebuilding of tooling binaries
-    gopath_cache:
-        folder: "${GOPATH}/bin"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
-
-    script:
-        - $SCRIPT_BASE/setup.sh
-        - $SCRIPT_BASE/vendor.sh
-
-
-docs_task:
-
-    depends_on:
-        - validate
-        - vendor
-
-    container:
-        dockerfile: contrib/cirrus/Dockerfile
-
-    timeout_in: 20m
-
-    # TODO: Making this task as not-requred to help speed up development given a new CI system
-    allow_failures: $CI == $CI
-
-    env:
-        CONTAINER: true
-
-    # Avoid downloading this stuff every time
-    gocache_cache:
-        folder: "${GOCACHE}"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
-
-    # Avoid needless rebuilding of tooling binaries
-    gopath_cache:
-        folder: "${GOPATH}/bin"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
-
-    # Avoid needless rebuilding of docs
-    gosrc_docs_cache:
-        folder: "${CIRRUS_WORKING_DIR}/docs"
-        fingerprint_script: $SCRIPT_BASE/cache_fingerprint.sh
+      image: golang:1.14
 
     script:
-        - $SCRIPT_BASE/setup.sh
-        - make docs
-        - ls -la docs
+        - make vendor
 
 
+# build binary and docs
 build_task:
 
-    depends_on:
-        - vendor
-        - validate
-
     # Uses VM via default gce_instance (above)
-
-    timeout_in: 10m
-
-    # TODO: Making this task as not-requred to help speed up development given a new CI system
-    allow_failures: $CI == $CI
+    gce_instance:
+        matrix:
+            image_name: $FEDORA_CACHE_IMAGE_NAME
```

Comment:
Sorry I'm late to the party...just a minor implementation note:  A matrix of one item is a very, very, sad matrix :cry:  This works too:

```yaml
    gce_instance:
        image_name:  $FEDORA_CACHE_IMAGE_NAME
```

Really the only reason to change this (drop the `matrix`) is to make life easier for Cirrus-CI n00bs reviewing or maintaining this configuration in the future.  Otherwise the technical impact is basically zero.